direnvrc

```diff
@@ -86,11 +86,50 @@ _nix_add_gcroot() {
   ln -fsn "$symlink" "/nix/var/nix/gcroots/per-user/$USER/$escaped_pwd"
 }
 
+# -n $(_nix_newer_than /path/to/cache /watched1 /watched2 /watched3)
+_nix_newer_than() {
+  local cache=$1
+  shift
+  local watched=("$@")
+  for f in "${watched[@]}"; do
+    if [[ "$f" -nt "$cache" ]]; then
+      echo "1"
+    fi
+  done
+}
+
+# _nix_add_watched /path/to/file
+declare -a _NIX_DIRENV_WATCHED=()
+_nix_add_watched() {
+  watch_file $1
+  _NIX_DIRENV_WATCHED+=($1)
+}
+
+_nix_newer_than_watched() {
+  _nix_newer_than $1 "${_NIX_DIRENV_WATCHED[@]}"
+}
+
+use_flake_if_supported() {
```

Comment:
This function could be also a separate project or gist that one can install on top of `nix-direnv` with `source_url`.